include/eosio/ship_protocol.hpp

```diff
@@ -808,6 +808,17 @@ namespace eosio { namespace ship_protocol {
 
    using resource_limits_config = std::variant<resource_limits_config_v0>;
 
+   struct finality_data {
+      uint32_t           major_version = {};
+      uint32_t           minor_version = {};
+      uint32_t           active_finalizer_policy_generation = {};
+      eosio::checksum256 action_mroot = {};
+      eosio::checksum256 base_digest = {};
+   };
```

Comment:
Shouldn't this include `proposed_finalizer_policy`

https://github.com/AntelopeIO/spring/blob/e1696d2dbbcd99879e303473572d5607be86cad5/libraries/chain/include/eosio/chain/block_state.hpp#L61-L61

Reply:
I built this from
https://github.com/AntelopeIO/spring/blob/8f65101b071768e8f213cd62e1a580f55aed882b/libraries/state_history/abi.cpp#L576-L581
which is lacking `proposed_finalizer_policy`. It does seem like this is an oversight in spring's abi.cpp

Reply:
Does look like that was missed.
https://github.com/AntelopeIO/spring/blob/3133bc1713a992139494eb3250c94a48336566a1/plugins/state_history_plugin/state_history_plugin.cpp#L290-L290
packs `finality_data_t` which contains `proposed_finalizer_policy`